product-web-app/tests/AuthService.test.ts

```diff
@@ -40,4 +47,49 @@ describe('AuthService', () => {
       }
     });
   });
+  describe('loginUser', () => {
+    it('should return error when pass wrong email', async () => {
+      const mock = new MockAdapter(axios);
+      mock.onPost('http://localhost:8080/api/auth/login', user).reply(400);
+
+      try {
+        await authService.login(login);
+      } catch (error) {
+        const errorMessage: string = error instanceof Error ? error.message : String(error);
+        expect(errorMessage).to.equal('Login failed! Please try again.');
+      }
```

Comment:
Move assertions to finally block to make sure they are always executed.
```suggestion
     } finally {
        expect(errorMessage).to.equal('Login failed! Please try again.');
      }
```